src/Core/Scheduler/Providers/PointCloudProvider.js

```diff
@@ -0,0 +1,282 @@
+        min = new THREE.Vector3().copy(cmin).add(xHalfLength);
+        max = new THREE.Vector3().copy(cmax).add(xHalfLength);
+    } else if (childIndex === 6) {
+        min = new THREE.Vector3().copy(cmin).add(xHalfLength).add(yHalfLength);
+        max = new THREE.Vector3().copy(cmax).add(xHalfLength).add(yHalfLength);
+    }
+
+    return new THREE.Box3(min, max);
+}
+
+
+function parseOctree(layer, hierarchyStepSize, root) {
+    return fetch(`${root.baseurl}/r${root.name}.hrc`, layer.fetchOptions).then(response => response.arrayBuffer()).then((blob) => {
+        const view = new DataView(blob);
+
+        const stack = [];
+
+        let offset = 0;
+
+        root.childrenBitField = view.getUint8(0); offset += 1;
+        root.numPoints = view.getUint32(1, true); offset += 4;
+        root.children = [];
+
+        stack.push(root);
+
+        while (stack.length && offset < blob.byteLength) {
+            const snode = stack.shift();
+            // look up 8 children
+            for (let i = 0; i < 8; i++) {
+                // does snode have a #i child ?
+                if (snode.childrenBitField & (1 << i) && (offset + 5) <= blob.byteLength) {
+                    const c = view.getUint8(offset); offset += 1;
+                    let n = view.getUint32(offset, true); offset += 4;
+                    if (n == 0) {
+                        n = root.numPoints;
+                    }
+                    const childname = snode.name + i;
+                    const bounds = createChildAABB(snode.bbox, i);
+
+                    let url = root.baseurl;
+                    if ((childname.length % hierarchyStepSize) == 0) {
+                        const myname = childname.substr(root.name.length);
+                        url = `${root.baseurl}/${myname}`;
+                    }
+                    const item = { numPoints: n, childrenBitField: c, children: [], name: childname, baseurl: url, bbox: bounds };
+                    snode.children.push(item);
+                    stack.push(item);
+                }
+            }
+        }
+
+        return root;
+    });
+}
+
+function readPositionAndColorBinFormat(ab, numPoints) {
+    const view = new DataView(ab);
+    let offset = 0;
+
+    numPoints = Math.min(numPoints, Math.floor(ab.byteLength / 16));
+
+    const positions = new Float32Array(3 * numPoints);
+    const colors = new Uint8Array(4 * numPoints);
+
+    for (let i = 0; i < numPoints; i++) {
+        positions[3 * i] = view.getUint32(offset + 0, true);
+        positions[3 * i + 1] = view.getUint32(offset + 4, true);
+        positions[3 * i + 2] = view.getUint32(offset + 8, true);
+
+        colors[4 * i] = view.getUint8(offset + 12);
+        colors[4 * i + 1] = view.getUint8(offset + 13);
+        colors[4 * i + 2] = view.getUint8(offset + 14);
+        colors[4 * i + 3] = 255;
+
+        offset += 16;
+    }
+    return [positions, colors];
+}
+
+function readPositionAndColorCustomBinFormat(ab, numPoints) {
```

Comment:
Actually, I plan to move all the file parsing to other files (BinLoader or Driver or whatever) when I'll refactor to support `pnts` format.